xplique/metrics/fidelity.py

```diff
@@ -407,8 +427,36 @@ def evaluate(self,
         Returns
         -------
         causal_score
-            Metric score, area over the deletion (lower is better) or insertion (higher is
-            better) curve.
+            Metric score (for interpretation, see score interpretation in the documentation).
+        """
+        scores_dict = self.detailed_evaluate(explanations)
+
+        # compute auc with trapeze
+        np_scores = np.array(list(scores_dict.values()))
+        auc = np.mean(np_scores[:-1] + np_scores[1:]) * 0.5
+
+        return auc
+
+    def detailed_evaluate(self,
+                          explanations: Union[tf.Tensor, np.ndarray]) -> Dict[int, float]:
+        """
+        Evaluate model performance for successive perturbations of an input.
+        Used to compute causal score for time series explanations.
+
+        Allow visualize the deletion and insertion curves.
+
+        Parameters
+        ----------
+        explanations
+            Explanation for the inputs, labels to evaluate.
+
+        Returns
+        -------
+        causal_score_dict
+            Dictionary of scores obtain for different perturbations
```

Comment:
What motivates the choice of a dictionnary rather than a list btw ? :)

Reply:
As aforementioned, this method main use is to allow visualization of insertion and deletion curves.
Thus keeping the number of features pertubed as keys of the dictionnary allow an easier implementation of the visualization and a more pertinent number of perturbation axis.